plugins/module_utils/azure_rm_common.py

```diff
@@ -1529,8 +1529,10 @@ def _get_profile(self, profile="default"):
             except Exception:
                 pass
 
-        if credentials.get('subscription_id'):
-            return credentials
+        if credentials.get('subscription_id') is None:
```

Comment:
The profile will be `None` if `credentials.get('subscription_id')` not `None`.   `credentials.get('subscription_id')` and `subscription_id` which one is the first priority?

Reply:
In order to not change the behavior I give `credentials.get('subscription_id')` the first priority

Reply:
`credentials.get('subscription_id') not None` not handled,  profile will returns as `None`

Reply:
@is-consulting  ping  `credentials.get('subscription_id') not None`   not handled. 
```suggestion
        if credentials.get('subscription_id') is None:
            if subscription_id is not None:
                credentials['subscription_id'] = subscription_id
                return credentials
        else:
            return credentials
```